src/approaches/grammar_search_invention_approach.py

```diff
@@ -619,10 +619,15 @@ def _evaluate_with_operators(self, predicates: FrozenSet[Predicate],
             init_atoms = utils.abstract(task.init,
                                         predicates | self._initial_predicates)
             objects = set(task.init)
+            ground_nsrts = task_plan_grounding(init_atoms, objects, strips_ops,
```

Comment:
I guess we can't really squeeze efficiency out of this call like we can for the `generate_heuristic` one below, right? Bc the predicates are always changing? But we only call this once per task instead of once per step in the task, so maybe okay?

Reply:
right, ideally we would find some optimizations for this too, but it's not obvious how because the predicates are changing